test/cli/install/bun-yarn.test.ts

```diff
@@ -0,0 +1,238 @@
+  expect(await getPackageJSONContents()).toMatchSnapshot();
+})
+
+const yarn_lock_files_from_internet = new Array<[org: string, repo: string]>();
+
+for (const org of await readdir("./resources/yarn")) {
+  for (const repo of await readdir(`./resources/yarn/${org}`)) {
+    yarn_lock_files_from_internet.push([org, repo]);
+  }
+}
+
+it(`should properly parse all ${yarn_lock_files_from_internet.length} yarn.lock files harvested from GitHub`, async () => {
+  const urls: string[] = [];
+  setHandler(dummyRegistry(urls, {}));
+  await makeBasicPackageJSON();
+  for (const [org, repo] of yarn_lock_files_from_internet) {
+    const yarn_lock = await file(`./resources/yarn/${org}/${repo}/yarn.lock`).text();
+    await writeFile(join(package_dir, "yarn.lock"), yarn_lock);
+    const { exited } = await command("install");
+    expect(await exited).toBe(0);
+  }
+})
+
+// Very, very, very slow! It's somewhat like a fuzzer
+/*
+it(`should not segfault for partial yarn lockfiles`, async () => {
+  for (const org of files) {
+    const urls: string[] = [];
+    setHandler(dummyRegistry(urls, {}));
+    await makeBasicPackageJSON();
+    for (const repo of await readdir(`./resources/yarn/${org}`)) {
+      const yarn_lock = await file(`./resources/yarn/${org}/${repo}/yarn.lock`).text();
+      for (let i = 0; ++i <= yarn_lock.length; ) {
+        await writeFile(join(package_dir, "yarn.lock"), yarn_lock.slice(0, i));
+        const { exited } = await command("i");
+        const exit_code = await exited;
+        expect(exit_code === 0 || exit_code === 1).toBe(true);
+      }
+    }
+  }
+})
+*/
+
+const test_yarn_source = `
+"@angular/build-tooling@https://github.com/angular/dev-infra-private-build-tooling-builds.git#414801b1253d48294087fed2bd4d7b07634c2b40", @angular/build-tooling@x:
+  version "0.0.0-7b31625b050e81dcdb575c22f4ed3991922b5bbc"
+  uid "414801b1253d48294087fed2bd4d7b07634c2b40"
+  resolved "https://github.com/angular/dev-infra-private-build-tooling-builds.git#414801b1253d48294087fed2bd4d7b07634c2b40"
+  integrity sha512-ApxoY5lTj1S0QFLdq5ZdTfdkIds1m3tma9EJOZpNVHRU9eCj2D/5+VFb5tlWsv9NHQ2S0XXkJjauFOAdfzT8uw==
+  dependencies:
+    "@angular/core" "^13.0.0 || ^14.0.0-0"
+    truth x
+  optionalDependencies:
+    "true" v1.2.3
+    yargs "^17.0.0"
+`
+
+const valid_start_and_end_states = [..."0111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111100111110000000000000000000000000000000000000000000011111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111001111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111110001111111111111111111111111111111111111111111111111111111110011111111110011111111111111111111111111111111111000000011111111111111111100"].map(c => +c);
```

Comment:
All the 1's are where it is invalid to stop, all the 0's are valid places to stop. These correspond to the code retrieved by `await exited`, which will be `0` if everything went swimmingly for Bun and `1` if there was an error that bubbled all the way up. Another possibility is a segfault, which is also verified by this as test as impossible.